lib/puppet/type/scheduled_task.rb

```diff
@@ -39,6 +39,10 @@
       # we are slash-insensitive. See #13009
       File.expand_path(value).gsub(/\//, '\\')
     end
+
+    def insync?(current)
+      provider.command_insync?(current, @should)
```

Comment:
What if `current` or `@should` are `:absent` or `nil`?

In the not-nil/absent case, is `@should` an array containing a single element or is it a String?